spec/helpers/companies_helper_spec.rb

```diff
@@ -151,21 +152,22 @@
 
     it 'returns all fields for member associated with company' do
       company = member.membership_applications[0].company
-      expect(show_address_fields(member, company))
+      expect(show_address_fields(member, company.addresses.first))
         .to match_array [ all_fields, true ]
     end
 
     it 'for visitor, returns fields consistent with address visibility' do
 
-      (0..Company::ADDRESS_VISIBILITY.length-1).each do |idx|
+      (0..Address::ADDRESS_VISIBILITY.length-1).each do |idx|
 
-        company.address_visibility = Company::ADDRESS_VISIBILITY[idx]
+        address.visibility = Address::ADDRESS_VISIBILITY[idx]
+        address.save!
 
-        fields, visibility = show_address_fields(visitor, company)
+        fields, visibility = show_address_fields(visitor, address)
 
         expect(visibility).to be false
 
-        case company.address_visibility
+        case address.visibility
         when 'none'
           expect(fields).to be nil
         else
```

Comment:
re: the next line in the file: `expect(fields).to match_array all_fields[idx, 5]`

I notice that the spec checks the range of `idx` to **`5`** but the code uses `4`.  Is that how it should be?  It just catches my eye, so I wanted to be sure it's ok.  

Seems that `companies_helper.rb` should really be refactored so that ` all_fields ` is a constant in that module (the helper) (or a value returned from a class method).  
We then could also use that constant in the tests instead of duplicating the array in the tests.  (there are + and - to doing that, of course.)  This could be done in a future PR if you feel it's out of scope of this one.

Reply:
Yes, thats correct.  

The spec access array elements using `Ary[start_index, length]` while the prod code uses `Ary[start_index, end_index]`

Reply:
The other idea (refactoring) should be deferred for now.  Also, I am leery of having the test, and the code-under-test, use the exact same constant.  Here, duplication might make sense if we want to discover a situation where the constant was changed but in a way that introduced a bug.  Part of the pluses and minuses you mention ...

Reply:
You're right.  Plus, since these 2 arrays don't really check the same thing (one _should_ use `4` and the other _should_ use `5`), it's a non-issue.